plugin/pkg/scheduler/factory/plugins.go

```diff
@@ -356,9 +359,23 @@ func getPriorityFunctionConfigs(names sets.String, args PluginFactoryArgs) ([]al
 			})
 		}
 	}
+	var totalPriority int
+	for _, config := range configs {
+		// Checks for overflow and returns errors.
+		// TODO: Need to check overflow for weight*MaxPriority.
+		if checkIntegerOverFlow(totalPriority, config.Weight*MaxPriority) {
+			return nil, fmt.Errorf("Total priority of priority functions has overflown")
+		}
+		totalPriority += config.Weight * MaxPriority
+	}
 	return configs, nil
 }
 
+// checkIntegerOverFlow checks if totalPriority can overflow after adding it with weight.
+func checkIntegerOverFlow(totalPriority int, weight int) bool {
```

Comment:
I agree with you but the overflow can happen at other places as well where weight is getting updated. So, I was suggesting that we should have checks there instead of here, as this is the last step in calculation. (I meant, we need to fail earlier than this.).